lib/tessel/wifi.js

```diff
@@ -98,8 +98,43 @@ Tessel.prototype.connectToNetwork = function(opts) {
       })
       .then(function() {
         // Then set the new credientials
-        logs.info('Credentials set!');
-        resolve();
+        return self.connection.exec(commands.commitWirelessCredentials());
```

Comment:
This the source of the one of the test failures—`commitWirelessCredentials` has already been called by `setWifiState` just before this. If you remove the entire `then`, that test no longer fails.